### ISSUE TITLE
If uptime and downtime overlap the replicaset will be scaled down

### ISSUE THREAD

**Issue by fbrnc** [hjacobs/kube-downscaler#28]
https://github.com/hjacobs/kube-downscaler/blob/1d5da010f3165c8781b25a77bc96f0360dbc29bf/kube_downscaler/scaler.py#L53

In situation where uptime and downtime overlap this line will evaluate to `is_uptime = false` resulting in the replica sets to be scaled down.
I wonder if this is excepted/wanted behavior. In case of doubt (e.g. conflicing configuration) I'd rather keep a replica set up and running instead of shutting it down :)

**Comment by hjacobs** [comment]
Yes, this behavior is expected, but we can obviously discuss what makes sense in general.

**Comment by fbrnc** [comment]
Maybe we could introduce an Apache-style (`order deny,allow`) flag that controls if uptime or downtime has a higher priority for overlapping configurations, defaulting to downtime having a higher priority for backwards-compatibility.
Also, currently being outside of an uptime window will automatically scale down even if no downtime window matches. This could be confusing. I think it would be clearer if everything would be untouched in case no configuration matches the current time.

**Comment by dh-harald** [comment]
@hjacobs I need this kind of functionality. We have a performance test environment, which is running only if we're doing performance tests... Otherwise the environment should be downscaled (replica 0). Just for sure, I'd like to set a downscale only rule after office hours... (if the environment needs againg, we'll make a new deployment anyway) How can I set that? Could you give me an example, if it's an excepted behavior?
